### PR TITLE
Remove IConversionExpression.IsExplicitInCode

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/22824. Since we have `IOperation.IsImplicit`, this property is redundant.

VSO : https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/515269

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 1]
```diff
@@ -794,13 +793,12 @@ private IConversionExpression CreateBoundAsOperatorOperation(BoundAsOperator bou
...
             SyntaxNode syntax = boundAsOperator.Syntax;
             Conversion conversion = boundAsOperator.Conversion;
-            bool isExplicit = true;
             bool isTryCast = true;
             bool isChecked = false;
             ITypeSymbol type = boundAsOperator.Type;
             Optional<object> constantValue = ConvertToOptional(boundAsOperator.ConstantValue);
-            bool isImplicit = boundAsOperator.WasCompilerGenerated;
-            return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit);
+            bool isImplicit = false; // as expressions can never be implicitly created
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: >// as expressions can never be implicitly created [](start = 37, length = 49)

That wouldn't be true for lowered tree. I would keep the original code.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 2]
```diff
@@ -517,7 +517,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
         Private Function CreateBoundTryCastOperation(boundTryCast As BoundTryCast) As IOperation
             Dim syntax As SyntaxNode = boundTryCast.Syntax
             Dim type As ITypeSymbol = boundTryCast.Type
-            Dim isImplicit As Boolean = boundTryCast.WasCompilerGenerated
+            Dim isImplicit As Boolean = False ' Try Cast can never be an implicit operation
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >Dim isImplicit As Boolean = False ' Try Cast can never be an implicit operation [](start = 12, length = 79)

Same as for C#, consider keeping the original code.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 3]
```diff
@@ -531,17 +531,16 @@ Namespace Microsoft.CodeAnalysis.Semantics
...
-            Return New LazyVisualBasicConversionExpression(operand, conversion, isExplicitCastInCode, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit)
+            Return New LazyVisualBasicConversionExpression(operand, conversion, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
         Private Function CreateBoundDirectCastOperation(boundDirectCast As BoundDirectCast) As IOperation
             Dim syntax As SyntaxNode = boundDirectCast.Syntax
             Dim type As ITypeSymbol = boundDirectCast.Type
             Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundDirectCast.ConstantValueOpt)
-            Dim isImplicit As Boolean = boundDirectCast.WasCompilerGenerated
+            Dim isImplicit As Boolean = False ' Direct Cast can never be an implicit operation
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >' Direct Cast can never be an implicit operation [](start = 46, length = 48)

Same here

**Review by AlekseyTs — APPROVED** [review]
LGTM with minor comments (iteration 1). I didn't review tests under assumption that all changes are removals of the property from the output.

**Review by mavasani — APPROVED** [review]

**Review by cston — APPROVED** [review]

**Comment by natidea** [comment]
Approved to merge via [515269](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/515269)

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Done.

**Pull request merged**